authors/apps/comments/tests/test_comment_history.py

```diff
@@ -0,0 +1,159 @@
+        slug = self.article_slug()
+        url = reverse("comments:post_comment", kwargs={'slug': slug})
+        res = self.client.post(url, data=comment, format="json")
+        data = res.data
+        comment_id = data["comment"]["id"]
+        fetch_url = reverse("comments:comment_history", kwargs={'slug':slug,'pk':comment_id})
+        response = self.client.get(fetch_url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn("Original comment", 
+                response.data["history"][0]["history_change_type"])
+
+    def test_get_history_update(self):
+        """
+        Test retriving a single comment
+        """
+        self.user_access()
+        self.posting_article(post_article)
+        slug = self.article_slug()
+        url = reverse("comments:post_comment", kwargs={"slug":slug})
+        res = self.client.post(url, data=comment, format="json")
+        data = res.data
+        comment_id = data["comment"]["id"]
+        update_url = reverse("comments:single_comment", kwargs={"slug":slug, "pk":comment_id})
+        self.client.put(update_url, data=update_comment, format="json")
+        fetch_url = reverse("comments:comment_history", kwargs={'slug':slug,'pk':comment_id})
+        response = self.client.get(fetch_url)
+        self.assertEqual(response.status_code, status.HTTP_200_OK)
+        self.assertIn("Updated",
+                    response.data["history"][1]["history_change_type"])
+
+    def test_get_history_wrong_user(self):
+        """
+        Test updating a comment 
+        """
+        self.user_access()
+        self.posting_article(post_article)
+        slug = self.article_slug()
+        url = reverse("comments:post_comment", kwargs={'slug': slug})
+        res = self.client.post(url, data=comment, format="json")
+        data = res.data
+        comment_id = data["comment"]["id"]
+        fetch_url = reverse("comments:comment_history", kwargs={'slug':slug,'pk':comment_id})
+        self.user_access2()
+        response = self.client.get(fetch_url)
+        self.assertEqual(response.status_code, status.HTTP_403_FORBIDDEN)
+        self.assertIn("You didn't create this comment Access Denied", 
+                response.data["message"])
+
+    def test_get_history_wrong_comment_id(self):
+        """
+        Test update comment with the same data 
+        """
+        self.user_access()
+        self.posting_article(post_article)
+        slug = self.article_slug()
+        url = reverse("comments:post_comment", kwargs={'slug': slug})
+        res = self.client.post(url, data=comment, format="json")
+        data = res.data
+        fetch_url = reverse("comments:comment_history", kwargs={'slug':slug,'pk':4})
+        response = self.client.get(fetch_url)
+        self.assertEqual(response.status_code, status.HTTP_404_NOT_FOUND)
+        self.assertIn("Not found", response.data["detail"])
+
+    # def test_update_a_comment_not_yours(self):
```

Comment:
Hey @wasibani-roy, I realize you have commented code,you could consider refactoring.